sc-memory/tests/sc-memory/common/sc-link.cpp

```diff
@@ -229,29 +231,81 @@ TEST_F(ScLinkTest, find_by_substr)
+  EXPECT_FALSE(ctx.FindLinksByContent("ton_content").empty());
+  EXPECT_FALSE(ctx.FindLinksByContent("cotents_25").empty());
 
   EXPECT_TRUE(ctx.FindLinksByContentSubstring("content1").size() == 1);
-  EXPECT_TRUE(ctx.FindLinksByContentSubstring("cont").size() == 3);
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("cont").size() == 2);
   EXPECT_TRUE(ctx.FindLinksByContentSubstring("content").size() == 2);
-  EXPECT_TRUE(ctx.FindLinksByContentSubstring("con").size() == 3);
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("on").size() == 3);
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("con").size() == 2);
   EXPECT_TRUE(ctx.FindLinksByContentSubstring("content2").empty());
-  EXPECT_TRUE(ctx.FindLinksByContentSubstring("contents").size() == 1);
-  EXPECT_TRUE(ctx.FindLinksByContentSubstring("contents_2").size() == 1);
-  EXPECT_TRUE(ctx.FindLinksByContentSubstring("contents_26").empty());
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("contents").empty());
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("cotents_2").size() == 1);
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("cotents_26").empty());
+
+  ctx.Destroy();
+}
+
+TEST_F(ScLinkTest, find_strings_by_substr)
+{
+  ScMemoryContext ctx(sc_access_lvl_make_min, "sc_links_content_changed");
+
+  ScAddr linkAddr = ctx.CreateLink();
+  EXPECT_TRUE(linkAddr.IsValid());
+  ScLink link1 = ScLink(ctx, linkAddr);
+  std::string str = "some coten";
+  EXPECT_TRUE(link1.Set(str));
+  EXPECT_TRUE(str == link1.GetAsString());
+
+  linkAddr = ctx.CreateLink();
+  EXPECT_TRUE(linkAddr.IsValid());
+  ScLink link2 = ScLink(ctx, linkAddr);
+  str = "ton_content";
+  EXPECT_TRUE(link2.Set(str));
+  EXPECT_TRUE(str == link2.GetAsString());
+
+  linkAddr = ctx.CreateLink();
+  EXPECT_TRUE(linkAddr.IsValid());
+  ScLink link3 = ScLink(ctx, linkAddr);
+  str = "content_tents_25";
+  EXPECT_TRUE(link3.Set(str));
+  EXPECT_TRUE(str == link3.GetAsString());
+
+  EXPECT_FALSE(ctx.FindLinksContentsByContentSubstring("some coten").empty());
+  EXPECT_FALSE(ctx.FindLinksContentsByContentSubstring("ton_content").empty());
+  EXPECT_FALSE(ctx.FindLinksContentsByContentSubstring("content_tents_25").empty());
+
+  auto strings = ctx.FindLinksContentsByContentSubstring("some coten");
+  EXPECT_TRUE(ctx.FindLinksContentsByContentSubstring("some coten").size() == 1);
+  EXPECT_TRUE(strings[0] == "some coten");
+
+  strings = ctx.FindLinksContentsByContentSubstring("cont");
+  EXPECT_TRUE(strings.size() == 2);
+  EXPECT_TRUE(std::find(strings.begin(), strings.end(), "ton_content") != strings.end());
+  EXPECT_TRUE(std::find(strings.begin(), strings.end(), "content_tents_25") != strings.end());
+
+  strings = ctx.FindLinksContentsByContentSubstring("25");
+  EXPECT_TRUE(strings.size() == 1);
+  EXPECT_TRUE(std::find(strings.begin(), strings.end(), "content_tents_25") != strings.end());
+
+  strings = ctx.FindLinksContentsByContentSubstring("co");
+  EXPECT_TRUE(strings.size() >= 3);
```

Comment:
Suggest to add comment in code to describe the reason why we have >3 results here